pallets/pallet-validator/src/lib.rs

```diff
@@ -0,0 +1,334 @@
+		/// Should be called when his node is synced and ready to start validating
+		#[pallet::weight(100_000_000)]
+		pub fn activate_validator(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			let address = ensure_signed(origin)?;
+
+			let mut validator = <Validator<T>>::get(&address)
+				.ok_or(DispatchError::from(Error::<T>::ValidatorNotFound))?;
+
+			ensure!(
+				validator.state != types::ValidatorRequestState::Validating,
+				Error::<T>::ValidatorValidatingAlready
+			);
+			ensure!(
+				validator.state == types::ValidatorRequestState::Approved,
+				Error::<T>::ValidatorNotApproved
+			);
+
+			// Update the validator request
+			validator.state = types::ValidatorRequestState::Validating;
+			<Validator<T>>::insert(address, &validator);
+
+			// Add the validator and rotate
+			substrate_validator_set::Pallet::<T>::add_validator(
+				frame_system::RawOrigin::Root.into(),
+				validator.validator_node_account.clone(),
+			)?;
+
+			Self::deposit_event(Event::ValidatorActivated(validator));
+
+			Ok(().into())
+		}
+
+		/// Change node validator account
+		/// In case the Validator wishes to change his validator node account
+		/// he can call this method with the new node validator account
+		/// this new account will be added as a new consensus validator
+		#[pallet::weight(100_000_000)]
+		pub fn change_node_validator_account(
+			origin: OriginFor<T>,
+			new_node_validator_account: T::AccountId,
+		) -> DispatchResultWithPostInfo {
+			let address = ensure_signed(origin)?;
+
+			let mut validator = <Validator<T>>::get(&address)
+				.ok_or(DispatchError::from(Error::<T>::ValidatorNotFound))?;
+
+			ensure!(
+				validator.state == types::ValidatorRequestState::Validating,
+				Error::<T>::ValidatorNotValidating
+			);
+
+			// Remove the old validator and rotate session
+			substrate_validator_set::Pallet::<T>::remove_validator(
+				frame_system::RawOrigin::Root.into(),
+				validator.validator_node_account.clone(),
+			)?;
+			Self::deposit_event(Event::ValidatorRemoved(validator.clone()));
+
+			// Set the new validator node account on the validator struct
+			validator.validator_node_account = new_node_validator_account.clone();
+			<Validator<T>>::insert(address, &validator);
+
+			// Add the new validator and rotate session
+			substrate_validator_set::Pallet::<T>::add_validator(
+				frame_system::RawOrigin::Root.into(),
+				new_node_validator_account.clone(),
+			)?;
+			Self::deposit_event(Event::ValidatorActivated(validator));
+
+			Ok(().into())
+		}
+
+		/// Bond an account to to a validator account
+		/// Just proves that the stash account is indeed under control of the validator account
+		#[pallet::weight(100_000_000)]
+		pub fn bond(
+			origin: OriginFor<T>,
+			validator: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResultWithPostInfo {
+			let stash = ensure_signed(origin.clone())?;
+
+			if <Bonded<T>>::contains_key(&stash) {
+				Err(Error::<T>::AlreadyBonded)?
```

Comment:
once bonded to a validator, always bonded to that validator account, even if the validator has already been removed.